### ISSUE TITLE
Some cookies are left untouched

### ISSUE THREAD

**Issue by franky47** [tmorin/ff-cpdn#1]
If I visit a cookie-heavy website such as https://oui.sncf, the extension removes localStorage contents correctly, but a lot of the cookies are left behind.

This is what I get in the console:
```
clear localStorage                                    tab-cleaner.js:4:9
clear sessionStorage                                  tab-cleaner.js:9:9
clear serviceWorker                                   tab-cleaner.js:22:9
clear serviceWorker - registrations 0                 tab-cleaner.js:25:13
clear caches                                          tab-cleaner.js:40:9
clear caches - keys 0                                 tab-cleaner.js:43:13
```
Interestingly, there is no mention of the log entries `cleanCookies -` nor `executeTabCleaner -`, which would mean that the cleanup tasks in background.js are not executed.

Service Workers are correctly unregistered (the website used as an example here does not have one, but I could verify that on a PWA).

**Comment by tmorin** [comment]
Hi @franky47 

Thanks for the feedback.

Cookies are not cleared within the JS context of the tab but from the background.js file of the extension. So you cannot see their logs in the tab's console but only from the broswer's console.

I checked https://oui.sncf, I guess you referred about domains: halc.iadvize.com and .demdex.net. From the [API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions), as far I understood, it is not possible to have the list of all cookies (whatever the domain) related to a given tab. The only thing available is the list of cookies related to an URL.

So in your case, only cookies related to oui.sncf will be cleared.

I will check if I can find a solution to discover cookies not related to the tab's URL.

Best regards

**Comment by franky47** [comment]
I have 3rd party cookie storage blocked so I don't have any of the ones you listed, but here are the ones that are left for me:

<img width="671" alt="oui sncf" src="https://user-images.githubusercontent.com/1174092/54869142-d07f1a00-4d94-11e9-9c16-acc802327f43.png">

Thanks for the explanation regarding the background logs !

**Comment by tmorin** [comment]
Hi @franky47 
I'm able to reproduce the issue but only in incognito mode. Is it your case?
Best regards

**Comment by franky47** [comment]
No, I'm in normal mode, but I have strict cookie settings:

<img width="673" alt="Screenshot 2019-03-23 at 19 33 36" src="https://user-images.githubusercontent.com/1174092/54870111-9d438780-4da2-11e9-88da-f0a3191e4962.png">

**Comment by tmorin** [comment]
Hi @franky47 ,
I may be find the issue. In fact the [context identifies](https://addons.mozilla.org/en-GB/firefox/addon/multi-account-containers/) were not handled by the extension. The next version, 1.3.2, solved it.
Best regards

**Comment by franky47** [comment]
After updating to 1.3.2, I still have the same outcome on that website.

On the extension console, I get the following:
```
cleanCookies - https://www.oui.sncf/ firefox-default   background.js:2:5
executeTabCleaner - https://www.oui.sncf/              background.js:32:5
cleanCookies - no contextual identity available        background.js:11:9
cleanCookies - discovered cookies 18                   background.js:19:5

... 18 cleanCookie events ommitted ...
```
The website stores 26 cookies, after cleanup of the 18 logged, the same 8 as in my previous comment remain.

**Comment by tmorin** [comment]
Hi @franky47,

I finally was able to get it ... I guess :)

In fact, the remaining cookies are not related to the tab's URL but to previous ones.

1. browses https://www.oui.sncf/monvoyage/recherche
2. browses https://www.oui.sncf/espaceclient/creer-compte
3. browses https://www.oui.sncf/train/train-avis
4. from console you will have cookies related to paths: /, /monvoyage, /espaceclient, /train
5. click on extension's button: only cookies with paths /monvoyage and /espaceclient are remaining

I published a new version: 1.4.0. This version now cleans cookies which have the domain of the tab's URL whatever their paths.

Best regards
